tests/AuthService.test.ts

```diff
@@ -0,0 +1,19 @@
+import { AuthService } from '../src/services/AuthService';
+
+describe('AuthService', () => {
+  let authService: AuthService;
+
+  beforeEach(() => {
+    authService = AuthService.getInstance();
+  });
+
+  test('should register a new user', async () => {
+    const user = await authService.registerUser('test@example.com', 'password123');
+    expect(user.email).toBe('test@example.com');
+  });
+
+  test('should authenticate user', async () => {
+    const result = await authService.authenticateUser('test@example.com', 'password123');
+    expect(result.token).toBeDefined();
+  });
+});
```

Comment:
Can we add more tests that cover the non-happy paths.  A few examples
- login and registration with invalid email and password
- what happens if the external auth service is down